src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs

```diff
@@ -819,6 +819,40 @@ async ValueTask ReleaseOnWriteCompletionAsync()
                 });
         }
 
+
+        [Fact]
+        public async Task WriteAsync_LocalAbort_Throws()
+        {
+            if (IsMockProvider)
+            {
+                // Mock provider does not support aborting pending writes via AbortWrite
+                return;
+            }
+
+            const int ExpectedErrorCode = 0xfffffff;
+
+            TaskCompletionSource waitForAbortTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            await RunBidirectionalClientServer(
+                clientStream =>
+                {
+                    return Task.CompletedTask;
+                },
+                async serverStream =>
+                {
+                    // write large enough buffer to make the call asynchronous
+                    byte[] buffer = ArrayPool<byte>.Shared.Rent(16 * 1024 * 1024);
+                    var writeTask = serverStream.WriteAsync(buffer);
```

Comment:
It still might happen so that WriteAsync task will be over by the time you call AbortWrite... I believe in other places we had an "infinite" loop of writes to make the test a bit more reliable

Reply:
The idea is that this should be enough data for MsQuic to actually wait until it is drained from the receiver's side before issuing the callback which completes the Task.

Can you point me to the place where we use the infinite loop? I'd like to see it.

Reply:
> wait until it is drained from the receiver's side

I believe msquic never waits for the receiver side. It will issue you the SEND_COMPLETE event as soon as it finishes copying to the internal buffer... have you seen it doing otherwise depending on the buffer size?

> the place where we use the infinite loop

https://github.com/dotnet/runtime/blob/e216e1300bad55af3472db9e64bc73eece2a488d/src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs#L600

https://github.com/dotnet/runtime/blob/e216e1300bad55af3472db9e64bc73eece2a488d/src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs#L672-L673

and I guess I also remembered this being in H3 tests
https://github.com/dotnet/runtime/blob/e216e1300bad55af3472db9e64bc73eece2a488d/src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs#L714-L719

Reply:
I see, but I wanted to explicitly test that the `SendResettableCompletionSource.CompleteException` happens on the pending Write. The `WriteForever loop would throw even before this change because there is an explicit check for aborted write inside `WriteAsync`...

So if we can't make this test reliable, then I would probably remove it entirely, as it does not really provide anything new.

Reply:
Thoughts, @ManickaP?

Reply:
If we didn't have this kind of check before -- and I guess we did not, since we didn't catch this bug from tests, only from looking at the code -- I would say let's leave the test, even if it works fully as intended, let's say, 90% of the time (and would not fail for the remaining 10%) it would still be enough to catch the bug if it will appear again.